### PR TITLE
RemoveVotes from templateView

### PR DESCRIPTION
**Pull request by GerhardPx**
# Pull Request

## Changes

Votes removed because they are no longer required, tests are adjusted and the 'required validation' for committeeSize is now removed. If no committeeSize is entered, the table header seats automatically becomes votes. 

## Reference

Issue: #SITZVERT_63

## Checklist

**Note**: If some checklist items are not relevant for your PR, just remove them.

- [x] Met all acceptance criteria of the issue
- [x] Added meaningful PR title and list of changes in the description
- [x] Wrote code and comments in English
- [x] Added tests
- [x] Removed waste on branch (e.g. `console.log`)
- [x] Considered and tested accessibility

## Summary by CodeRabbit

* **New Features**
  * Single "seats or votes" input across group rows; header icon and label switch contextually. Summary row consolidated to show total (and "von …" when seats apply).
  * Add-row uses a contextual maximum for the single input.

* **Bug Fixes**
  * Prevents false "too many seats" warnings when main organ size is zero or unset.
  * Main organ size input is no longer required; min/max validation retained.

* **Tests**
  * Test data and specs updated to the single-field model.

* **Documentation**
  * Clarified input guidance: if main organ size is set, enter seats; otherwise inputs are treated as votes.

### CODE REVIEW

File: sitzverteilung-frontend/src/components/basedata/BaseDataForm.vue [hunk 1]
```diff
@@ -22,7 +22,6 @@
       <v-col>
         <v-number-input
           v-model="baseData.committeeSize"
-          :rules="[FieldValidationRules.Required]"
```
[the remark this answers is not in the document]
Comment (devtobi) [reply sitzverteilung-frontend/src/components/basedata/BaseDataForm.vue:25]: Und wieso ist in dieser Komponente zweimal ein NumberInput mit `baseData.committeeSize` als Model?

File: sitzverteilung-frontend/src/composables/useGroupStatistics.ts [hunk 2]
```diff
@@ -16,15 +16,10 @@ export const useGroupStatistics = (
...
-  const totalVotes = computed(() => {
-    return groups.value.reduce((sum, group) => sum + (group.votes ?? 0), 0);
-  });
 
   // validations
   const safeExpectedSeats = computed(() => expectedSeats.value ?? 0);
-  const maximumGroups = computed(() =>
-    Math.min(limitGroups.value, safeExpectedSeats.value)
-  );
+  const maximumGroups = computed(() => Math.min(limitGroups.value));
```
Comment (devtobi) [inline sitzverteilung-frontend/src/composables/useGroupStatistics.ts:22]: Der Aufruf so macht keinen Sinn, da Math.min eigentlich immer zwei Parameter braucht. Siehe aber auch vorherige Anmerkung zur Überarbeitung der Logik.

File: sitzverteilung-frontend/src/views/CalculationView.vue [hunk 3]
```diff
@@ -44,7 +44,6 @@
       :limit-name="LimitConfiguration.limitName"
       :limit-groups="LimitConfiguration.limitGroups"
       :limit-committee-size="LimitConfiguration.limitCommitteeSize"
-      :limit-votes="LimitConfiguration.limitVotes"
```
Comment (devtobi) [inline sitzverteilung-frontend/src/views/CalculationView.vue:47]: Würde ich beibehalten, siehe auch andere Anmerkungen hierzu.

File: sitzverteilung-frontend/src/views/TemplateView.vue [hunk 4]
```diff
@@ -199,7 +199,6 @@
           :limit-name="LimitConfiguration.limitName"
           :limit-groups="LimitConfiguration.limitGroups"
           :limit-committee-size="LimitConfiguration.limitCommitteeSize"
-          :limit-votes="LimitConfiguration.limitVotes"
```
Comment (devtobi) [inline sitzverteilung-frontend/src/views/TemplateView.vue:202]: Würde ich beibehalten, siehe auch andere Anmerkungen hierzu.

File: sitzverteilung-frontend/tests/TestData.ts [hunk 5]
```diff
@@ -10,17 +10,14 @@ export function getTestBaseData(): BaseData {
       {
         name: "Testgroup 1 ä",
         committeeSeats: 10,
```
Comment (devtobi) [inline sitzverteilung-frontend/tests/TestData.ts:12]: Anpassung auf `seatsOrVotes`, siehe auch andere Anmerkungen.

File: sitzverteilung-frontend/tests/TestData.ts [hunk 6]
```diff
@@ -125,9 +133,6 @@ export function getTestBaseDataInputLimit(): BaseData {
     ),
     groups: Array.from({ length: LimitConfiguration.limitGroups }, () => ({
       name: generateUniqueString(LimitConfiguration.limitName),
-      votes: generateUniqueNumber(
-        LimitConfiguration.limitVotes.toString().length
-      ),
       committeeSeats: generateUniqueNumber(
         LimitConfiguration.limitCommitteeSize.toString().length
       ),
```
Comment (devtobi) [inline sitzverteilung-frontend/tests/TestData.ts:138]: Da in dem Feld jetzt theoretisch Sitze oder Votes möglich sind, müsste man eigentlich mit dem Limit der Votes testen, da dieses Feld deutlich länger als die Sitze im Falle der Angabe eines Hauptorgans ist. Daher muss hier die Methode zur Generierung des Randfalls angepasst werden.

File: sitzverteilung-frontend/tests/TestData.ts [hunk 7]
```diff
@@ -105,12 +97,28 @@ export function getTestBaseDataNotEnoughSeats(): BaseData {
...
         name: "Testgroup 2",
         committeeSeats: 3,
-        votes: 200,
+      },
+    ],
+    unions: [],
+  };
+}
+
+export function getTestBaseDataSeatsAreNull(): BaseData {
```
Comment (devtobi) [inline sitzverteilung-frontend/tests/TestData.ts:110]: Wieso benötigen wir diese neue Methode. Hat sich durch die Änderung ein neuer Edge-Case ergeben, der getestet werden muss?

File: sitzverteilung-frontend/tests/TestData.ts [hunk 8]
```diff
@@ -152,7 +157,6 @@ export function getTestBaseDataInputTooLarge(): BaseData {
     committeeSize: generateUniqueNumber(3),
     groups: Array.from({ length: 18 }, () => ({
       name: generateUniqueString(999),
-      votes: generateUniqueNumber(9),
```
Comment (devtobi) [inline sitzverteilung-frontend/tests/TestData.ts:155]: Sollte so passen, da das Objekt durch die Strings der Länge 999 auf jeden Fall noch zu lang wird.

File: sitzverteilung-frontend/src/composables/useGroupStatistics.ts [hunk 9]
```diff
@@ -16,15 +16,10 @@ export const useGroupStatistics = (
       0
     );
   });
-  const totalVotes = computed(() => {
-    return groups.value.reduce((sum, group) => sum + (group.votes ?? 0), 0);
-  });
```
Comment (devtobi) [inline sitzverteilung-frontend/src/composables/useGroupStatistics.ts:21]: Sollten wir dann nicht ein `totalSeatsOrVotes` einführen, um beides miteinander zu kombinieren?

File: sitzverteilung-frontend/tests/composables/useGroupStatistics.spec.ts [hunk 10]
```diff
@@ -18,42 +19,38 @@ describe("useGroupStatistics composable", () => {
     const expectedSeats = ref(testBaseData.committeeSize);
     const limitGroups = ref(10);
 
-    const { amountOfGroups, totalSeats, totalVotes } = useGroupStatistics(
+    const { amountOfGroups, totalSeats } = useGroupStatistics(
```
Comment (devtobi) [inline sitzverteilung-frontend/tests/composables/useGroupStatistics.spec.ts:22]:
```suggestion
    const { amountOfGroups, totalSeatsOrVotes } = useGroupStatistics(
```

File: sitzverteilung-frontend/tests/composables/useGroupStatistics.spec.ts [hunk 11]
```diff
@@ -18,42 +19,38 @@ describe("useGroupStatistics composable", () => {
...
     expect(amountOfGroups.value).toBe(3);
     expect(totalSeats.value).toBe(60);
-    expect(totalVotes.value).toBe(350);
   });
 
   test.each([
     [getTestBaseData(), false, false, false],
     [getTestBaseDataEmptyGroups(), false, false, false],
-    [getTestBaseDataTooManyGroups(), true, false, false],
+    [getTestBaseDataTooManyGroups(), false, false, false],
```
Comment (devtobi) [inline sitzverteilung-frontend/tests/composables/useGroupStatistics.spec.ts:35]: So macht doch der Testfall keinen Sinn mehr? Es sollen ja zu viele Gruppen sein, daher bitte auf `true` lassen und ggf. die Testdaten anpassen.

File: sitzverteilung-frontend/tests/composables/useGroupStatistics.spec.ts [hunk 12]
```diff
@@ -18,42 +19,38 @@ describe("useGroupStatistics composable", () => {
...
     [getTestBaseDataTooManySeats(), false, true, false],
-    [getTestBaseDataUndefinedTooManySeats(), true, true, false],
+    [getTestBaseDataUndefinedTooManySeats(), false, true, false],
```
Comment (devtobi) [inline sitzverteilung-frontend/tests/composables/useGroupStatistics.spec.ts:37]: Selbiges wie Kommentar zuvor.

File: sitzverteilung-frontend/tests/composables/useGroupStatistics.spec.ts [hunk 13]
```diff
@@ -18,42 +19,38 @@ describe("useGroupStatistics composable", () => {
...
     [getTestBaseDataNotEnoughSeats(), false, false, true],
+    [getTestBaseDataSeatsAreNull(), false, false, true],
```
Comment (devtobi) [inline sitzverteilung-frontend/tests/composables/useGroupStatistics.spec.ts:39]: Welchen Fall deckt dieser Test ab?

File: sitzverteilung-frontend/tests/composables/useGroupStatistics.spec.ts [hunk 14]
```diff
@@ -18,42 +19,38 @@ describe("useGroupStatistics composable", () => {
...
   ])(
     "correctly calculates validation",
-    (
-      testBaseData,
-      expectedTooManyGroups,
-      expectedSeatsTooHigh,
-      expectedSeatsTooLow
-    ) => {
+    (testBaseData, expectedTooManyGroups) => {
```
Comment (devtobi) [inline sitzverteilung-frontend/tests/composables/useGroupStatistics.spec.ts:42]: Wie von CodeRabbit bereits angemerkt, passt die Methodensignatur nicht mehr zu der Anzahl übergegeber Parameter im parametrisierten Test. Bitte korrigieren.

File: sitzverteilung-frontend/tests/composables/useGroupStatistics.spec.ts [hunk 15]
```diff
@@ -18,42 +19,38 @@ describe("useGroupStatistics composable", () => {
...
       const limitGroups = ref(10);
       const expectedSeats = ref(testBaseData.committeeSize);
 
-      const { isTooManyGroups, isSeatsTooLow, isSeatsTooHigh } =
-        useGroupStatistics(groups, limitGroups, expectedSeats);
+      const { isTooManyGroups } = useGroupStatistics(
+        groups,
+        limitGroups,
+        expectedSeats
+      );
```
Comment (devtobi) [inline sitzverteilung-frontend/tests/composables/useGroupStatistics.spec.ts:51]: Wieso wurden die anderen Attribute rausgenommen, sollten die nicht auch überprüft werden?
